sdks/python/apache_beam/transforms/core.py

```diff
@@ -1423,15 +1423,21 @@ def _check_fn_use_yield_and_return(fn):
     source_code = _get_function_body_without_inners(fn)
     has_yield = False
     has_return = False
+    if len(source_code)==0:
+      return None
     for line in source_code.split("\n"):
       if line.lstrip().startswith("yield ") or line.lstrip().startswith(
           "yield("):
         has_yield = True
       if line.lstrip().startswith("return ") or line.lstrip().startswith(
           "return("):
         has_return = True
+      if line.lstrip().startswith("return None"):
```

Comment:
The warn could also be placed here instead, eliminating the need for the `elif` block.